### PR TITLE
Copter: pre-arm check of deprecated FS_GCS_ENABLE = 2 (Continue in Auto)

### PR DESCRIPTION
**Pull request by rmackay9**
This PR makes two changes related to the FS_GCS_ENABLE and FS_OPTION parameters:

- adds a pre-arm check if FS_GCS_ENABLE == 2 (continue in Auto) to catch users who may have set this to "2".  Below is a screen shot captured while testing this check
image

- FS_OPTIONS defaults to "16" (was "0") which means, "Continue if in pilot controlled modes on GCS failsafe".  This was discussed on a weekly dev call about 4 to 6 weeks ago and we decided this what most users would expect.  This has been tested in SITL to ensure that new installations of 4.1 will have the value set to "16" and also that upgrades from 3.6.x (which did not have the FS_OPTIONS parameter) get set correctly as part of param conversion.

Below is a screenshot of the parameter conversion (from 3.6.x to 4.1) result when FS_GCS_ENABLE = 2 (in Copter-3.6.x).  This is correctly converted to 18 (2+16) in Copter-4.1.
image

This PR closes issue: https://github.com/ArduPilot/ardupilot/issues/15712

### CODE REVIEW

**Review by bnsgeyer — APPROVED** [review]
@rmackay9 thanks for including the default setting of the failsafe to not cause a GCS failsafe in a pilot controlled mode.  It was on my list of things to do.  I will close my PR.

File: ArduCopter/AP_Arming.cpp [hunk 1]
```diff
@@ -159,6 +159,10 @@ bool AP_Arming_Copter::parameter_checks(bool display_failure)
                 return false;
             }
         }
+        if (copter.g.failsafe_gcs == FS_GCS_ENABLED_CONTINUE_MISSION) {
+            // FS_GCS_ENABLE == 2 is deprecated
+            check_failed(ARMING_CHECK_PARAMETERS, display_failure, "FS_GCS_ENABLE=2 deprecated, see FS_OPTIONS");
```
Comment (tridge) [inline ArduCopter/AP_Arming.cpp:164]: change "deprecated" to "removed"

Comment (tridge) [reply ArduCopter/AP_Arming.cpp:164]: should also fix in param description to be removed

**Review by tridge — DISMISSED** [review]

File: ArduCopter/AP_Arming.cpp (hunk 1 above)
Comment (rmackay9) [reply ArduCopter/AP_Arming.cpp]: replaced "deprecated" with "removed" in param descriptions for FS_GCS_ENABLE and FS_THR_ENABLE

**Pull request merged**